tests/test_in_out.py

```diff
@@ -180,3 +180,34 @@ def test_spinner_write_race_condition(capsys):
     assert "aaaa" in out  # spinner text is present
     assert "bbbb" in out  # write() text is present
     assert not re.search(r"aaaa[^\rb]*bbbb", out)
+
+
+def test_spinner_hidden_with_context_manager(capsys):
+    HIDDEN_START = "hidden start"
+    HIDDEN_END = "hidden end"
+    sp = yaspin(text="foo")
+    sp.start()
+
+    # first hidden hides the spinner
+    with sp.hidden():
+        assert sp._hide_spin.is_set()
+        sp.write(HIDDEN_START)
+
+        # test nested hiding
```

Comment:
What's the use case for nesting contextmanagers?

Reply:
The basic idea of supporting nesting is that we don't have to check if the spinner is hidden before we use it in a `with` statement. E.g. the spinner might be used in a bigger cli application, where it is passed around to different helper methods. The helper methods might not know the current state of the spinner (another helper method might already have hidden the spinner), but might want to make sure that the spinner is hidden for their use-case. If we don't support nesting then a nested `with` might unintentionally unhide the spinner on leave.